source/develop/release-management/features/network/openvswitch/native-openvswitch.html.md

```diff
@@ -0,0 +1,200 @@
+  - When upgrading from legacy to OVS, there is a need to tear down bridges and configure OVS in one transaction and during one rollback.
+
+### VDSM API
+Minimal changes in current API verbs are required:
+
+- setupNetworks: Per network, add a switch_type key.
+  - Currently it can have two values: ‘legacy’ and ‘ovs’.
+  - If not explicitly set, assume ‘legacy’. (for compatability with Engine 3.6)
+- oVirt currently differentiates between VM (bridged) networks and non-VM networks due to historical performance benefit of non-bridged networks for non-VM payloads (such as storage and migration). This differentiation is moot with OVS, but in order to keep Engine oblivious to the change, VDSM should lie to it, and report bridged=False for networks that requested it. Dropping the differentiation from Engine should happen in a later stage.
+
+### Upgrade from native to OVS networking
+Transition between switch types must be supported for the host as a whole. Nevertheless, if the user has defined his own networking configuration which is not controlled by VDSM, it should coexist with VDSM controlled networks.
+
+Mixing between native and OVS controlled networks is not initially supported, therefore, when a network is marked with a different type from the existing, a validation check should make sure that all other networks are marked for the same network type.
+
+The following sequence describes the transition steps @Engine:
+
+- Vdsm on hosts is upgraded to an OVS-supporting version (4.0). No change takes place to preexisting network configuration, which would still be bridge-based.
+- When all active hosts in the cluster have been upgraded, the user should upgrade the cluster to version 4.0 where OVS is supported.
+- The default switch type for the cluster should be set by the user to OVS (for new hosts).
+  - New created 4.0 clusters should have OVS as the default switch.
+- Update individual hosts to transition to OVS: 
+(TBD: one or all of the options below should be chosen)
+  - The host networking can be changed when no VM/s are connected to it. With this option oVirt must support VM migration between different switch types.
+  - The host networking can be changed in-place without the need to migrate the VM/s. With this option, migration between hosts that have different networking types is not needed (but can coexist).
+Neglible service interruption to the VM networking is expected.
+
+The following sequence describes the transition steps @Host:
+- Receive a setupNetworks call.
+- Detect that a network has been marked with a type that is different from the current.
+- Validate that all existing networks exist in the command and that all are marked with the same type.
+  - We would like to perform an atomic upgrade, and not to worry about codependent networks with differing switch types. 
+- Tear down the pre-existing networks and bondings..
+- Create networks.
+
```

Comment:
please add a note: Engine user may decide to take the reverse transition, from OVS to legacy, as well.